gpu/base/executor.cpp

```diff
@@ -46,34 +46,59 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 namespace gko {
 
 
-void CpuExecutor::raw_copy_to(const GpuExecutor *, size_type num_bytes,
+class device_guard {
```

Comment:
nit: since this is an internal structure, make sure it has internal linkage by wrapping it into an [anonymous namespace](http://en.cppreference.com/w/cpp/language/namespace#Unnamed_namespaces).